### ISSUE TITLE
Regression in editor buttons clipping last letter under certain circumstances

### ISSUE THREAD

**Issue by GuyUnger** [godotengine/godot#89122]
### Tested versions

happens in v4.3.dev4
does not happen in 4.3.dev3, 4.3.dev2 or 4.3.dev1 (i tested)

### System information

Windows 11 -  4.3.dev4.official [df78c0636]

### Issue description

Using the theme by passivestar https://github.com/passivestar/godot-minimal-theme with the editor on 75% scaling the last letters in the top buttons 2d, 3d, script, assetlib get hidden on hover. Apparently this was an issue before ( https://github.com/godotengine/godot/issues/85449 ) under the exact same circumstances ( https://github.com/godotengine/godot/issues/88066 ) and was solved, probably with https://github.com/godotengine/godot/pull/87335
https://github.com/godotengine/godot/pull/86378 might have re-introduced this issue but not certain

### Steps to reproduce

use the theme by passivestar ( https://github.com/passivestar/godot-minimal-theme ) and set scaling to 75%

### Minimal reproduction project (MRP)

this happens in editor with any project

**Comment by passivestar** [comment]
Also makes the icon shift but I assume that's because of the missing letter

https://github.com/godotengine/godot/assets/60579014/c61ca7cf-e11c-445c-a8c8-3aced853f96e

**Comment by akien-mga** [comment]
Related to https://github.com/godotengine/godot/pull/88652#issuecomment-1969682222, CC @RobProductions @Calinou

**Comment by Rindbee** [comment]
https://github.com/godotengine/godot/blob/f2045ba822bff7d34964901393581a3117c394a9/editor/themes/editor_theme_manager.cpp#L2448-L2454

`EDSCALE` was not considered when merging themes.

**Comment by RobProductions** [comment]
I believe this comes from the "Base spacing" variable which as you can see here breaks the stylebox when it's set to < 4:

https://github.com/godotengine/godot/assets/11141862/5c40b8a8-c429-4a78-a3c7-41c30f5bc238

It does seem to be limited to external themes though (or at least the one linked here which I tried) and thankfully doesn't break on default Godot UI:

https://github.com/godotengine/godot/assets/11141862/e604e6d7-9884-45ba-8000-24ec3f325edc

From what I can tell Editor Scale has no bearing on it as both videos were recorded with 100% scale, but I could be wrong. The hover stylebox margins did change in #88652 as you can see here: 

https://github.com/godotengine/godot/pull/88652/files

Though that was meant to fix the "draw extra borders" margin that should've been passed down to that. Could be breaking because the new "MainScreenButton" variation needs the same margin treatment as the flat buttons but I believe when last I tested that it also appeared cut off, hence why it has different margins from the `menu_transparent_style` that it stems from. It remained that way because I couldn't find any case that breaks it, but with external themes it seems there is an issue.

Took a while to assemble all that because I did a fresh compilation of Godot on a new machine but now that I'm able to reproduce it I can try testing out some fixes, just might take a bit of trial and error. I don't know exactly what's wrong yet but hopefully that narrows down the search, apologies for not catching it earlier!

**Comment by Rindbee** [comment]
It is mainly caused by the use of different minimum size styleboxes for the same button in different statuses.

In this MRP, the minimum sizes of styleboxes are different in `hover` and `normal` status.

![1](https://github.com/godotengine/godot/assets/30386067/4ba980d0-0577-45a6-9755-feac28d2ed9a)

Since `MainScreenButton` lacks the stylebox used in `hover` status, it will fall back to `Button`.

![2](https://github.com/godotengine/godot/assets/30386067/c437b685-db9c-4496-afcf-1359519f82ee)

In addition, `interface/editor/display_scale` will not affect custom themes (https://github.com/godotengine/godot/issues/89122#issuecomment-1976439468), custom theme items are usually values assuming scale is 100%. Therefore, this issue will occur when `interface/editor/display_scale` is not 100% and a custom theme that is not fully covered is used.

Finally, I'm confused by the code that calls `set_content_margin()` with the results of `get_margin()` and `get_border_width()`.

https://github.com/godotengine/godot/blob/a07dd0d6a520723c4838fb4b65461a16b7a50f90/editor/themes/editor_theme_manager.cpp#L1822-L1824

https://github.com/godotengine/godot/blob/a07dd0d6a520723c4838fb4b65461a16b7a50f90/scene/resources/style_box.cpp#L78-L86

https://github.com/godotengine/godot/blob/a07dd0d6a520723c4838fb4b65461a16b7a50f90/scene/resources/style_box_flat.cpp#L35-L38

https://github.com/godotengine/godot/blob/a07dd0d6a520723c4838fb4b65461a16b7a50f90/scene/resources/style_box_flat.cpp#L82-L85

**Comment by RobProductions** [comment]
> It is mainly caused by the use of different minimum size styleboxes for the same button in different statuses.
> Since `MainScreenButton` lacks the stylebox used in `hover` status, it will fall back to `Button`.

Good catch! I defined the `hover` stylebox in the theme variation with the correct margins (same fix as what I showed above in #88652 ) and that seemed to solve the problem:

https://github.com/godotengine/godot/assets/11141862/3eebf637-a65c-42ca-9aae-9d9a1e6d6783

> In addition, `interface/editor/display_scale` will not affect custom themes ([#89122 (comment)](https://github.com/godotengine/godot/issues/89122#issuecomment-1976439468)), custom theme items are usually values assuming scale is 100%. Therefore, this issue will occur when `interface/editor/display_scale` is not 100% and a custom theme that is not fully covered is used.

You're right, I do see an issue at 75% scale on the custom theme:

![image](https://github.com/godotengine/godot/assets/11141862/fd07039f-4c23-472f-a827-6539099f7f11)

But I believe it's only because internally it sets the margin value based on EDSCALE as you can see here:

https://github.com/godotengine/godot/blob/a07dd0d6a520723c4838fb4b65461a16b7a50f90/editor/themes/editor_theme_manager.cpp#L129

Hence the issue still stems from margins and correcting them as shown in the video allows it to stay constrained on all scale values too, so as far as I can tell this should resolve it for now.

> If it wants to use `content_margin[p_side]` here, why not use `get_content_margin()` instead.

The code I wrote that you see there currently mirrors an existing pattern that was there previously which I guess was a little more convoluted than it needed to be, but I tested it and you're right, it will work with `get_content_margin()` too! I updated those lines to use it, thanks for pointing that out!

There are just two things I need to figure out before I submit my PR, one is why the hover highlights on the main buttons seem to lose its rounded edge and another is cleaning up the bottom bar `pressed` style which also shows clipping at times. The bottom bar issue is puzzling because it seems that assigning a stylebox override is what causes these margins to get messed up. Perhaps the custom theme wipes that out somehow and it just needs its own theme variation like the main screen buttons, but when I tried that it still cuts off. Will update when I learn more and I think like I'm pretty close to having this ready now :)
